tests/test_create_model.py

```diff
@@ -254,15 +254,22 @@ def test_create_model_field_and_model_title():
 
 
 def test_create_model_field_description():
-    m = create_model('M', a=(str, Field(description='descr')))
+    m = create_model('M', a=(str, Field(description='descr')), __doc__='Some doc')
     assert m.model_json_schema() == {
         'properties': {'a': {'description': 'descr', 'title': 'A', 'type': 'string'}},
         'required': ['a'],
         'title': 'M',
         'type': 'object',
+        'description': 'Some doc',
     }
 
 
+def test_create_model_with_doc():
+    model = create_model('FooModel', foo=(str, ...), bar=(int, 123), __doc__='The Foo model')
+    assert model.__name__ == 'FooModel'
+    assert model.__doc__ == 'The Foo model'
+
+
```

Comment:
Awesome, thanks for adding some tests. Could you add one more test that checks that if a model is created with a `__base__` (some other model), the new model inherits the docstring from the `__base__` model?

Reply:
No problem! I have added a test for this [here](https://github.com/pydantic/pydantic/pull/7863/commits/291dd3ad68f3fb1b6023acc6bd36d868bd717824).